lti_consumer/lti_xblock.py

```diff
@@ -93,7 +93,7 @@
     "/projects/open-edx-building-and-running-a-course/en/latest/exercises_tools/lti_component.html"
     "'>"
 )
-RESULT_SERVICE_SUFFIX_PARSER = re.compile(r"^user/(?P<anon_id>\w+)", re.UNICODE)
+RESULT_SERVICE_SUFFIX_PARSER = re.compile(r"^user/(?P<anon_id>[\w-]+)", re.UNICODE)
```

Comment:
This adds support for UUID user IDs. Otherwise, it lops off part of the user ID.

Reply:
this must have been maddening to find